stl/inc/valarray

```diff
@@ -591,22 +603,22 @@ void swap(valarray<_Ty>& _Left, valarray<_Ty>& _Right) noexcept {
 
 template <class _Ty>
 _NODISCARD _Ty* begin(valarray<_Ty>& _Array) {
-    return &_Array[0];

```

Comment:
I would prefer to move them inline, but I am not sure whether there are any ABI concerns rjere

Reply:
We conventionally don't declare function templates `inline` (if I'm interpreting "move them inline" correctly) since (1) it isn't needed to satisfy ODR, and (2) the inliner almost ignores the presence of `inline` as a hint.

Reply:
I think they mean simply moving the definitions up into the class, which I think is indeed a bit cleaner. Although if I'm not wrong this also causes the function templates to be instantiated whenever the valarray class is, which I'm not sure is a concern?

Reply:
Defining these in the class body (absent redeclarations at namespace scope) would make them hidden friends. This is better design and would have been a great idea in 2011 when `valarray`'s `begin` and `end` overloads were standardized. Today it would likely break code in the wild which uses `std::begin(meow)` / `std::end(meow)` for a valarray `meow`, and `valarray` is just not worth the trouble.

TLDR: Moving them into the class body gains us nothing since we'd need to redeclare them at namespace scope.

Reply:
>  (2) the inliner almost ignores the presence of `inline` as a hint

To be precise, for /Ob1 it doesn't ignore inline.